crates/runtime/src/object_store_registry.rs

```diff
@@ -0,0 +1,94 @@
+use std::{collections::HashMap, sync::Arc};
+
+use datafusion::{
+    error::DataFusionError,
+    execution::{
+        object_store::{DefaultObjectStoreRegistry, ObjectStoreRegistry},
+        runtime_env::{RuntimeConfig, RuntimeEnv},
+    },
+};
+use object_store::{aws::AmazonS3Builder, ObjectStore};
+use url::{form_urlencoded::parse, Url};
+
+#[allow(clippy::module_name_repetitions)]
+#[derive(Debug, Default)]
+pub struct SpiceObjectStoreRegistry {
+    inner: DefaultObjectStoreRegistry,
+}
+
+impl SpiceObjectStoreRegistry {
+    #[must_use]
+    pub fn new() -> Self {
+        SpiceObjectStoreRegistry::default()
+    }
+
+    #[allow(clippy::unused_self)]
+    fn get_feature_store(&self, url: &Url) -> datafusion::error::Result<Arc<dyn ObjectStore>> {
```

Comment:
```suggestion
    fn get_feature_store(url: &Url) -> datafusion::error::Result<Arc<dyn ObjectStore>> {
```

And then to call it, use `Self::get_feature_store(url)?;` below

Reply:
I left it so that in future we can pass in some context here - E.g. configuration stuff.

I guess we can refactor it to Self:: and change it when it is needed.

Reply:
Yeah - I think that is cleaner for now (refactor to use Self::)